### ISSUE TITLE
Assertion failed when importing CSV with string list

### ISSUE THREAD

**Issue by rfdavid** [kuzudb/kuzu#1352]
To benchmark #1327 for string list, I created the following node table and imported a CSV with randomly generated values:

`CREATE NODE TABLE Preference(name STRING, pref STRING[],  PRIMARY KEY (name));`

When trying to match the list, I'm getting an assertion failed:

```
➜  shell git:(master) ./kuzu_shell -i db
Opened the database at path: db
Enter ":help" for usage hints.
kuzu> match (p:Preference) return p.pref;
Assertion failed: (srcEndIdx < src.size), function copyListRecursiveIfNested, file in_mem_overflow_buffer_utils.cpp, line 31.
[1]    89962 abort      ./kuzu_shell -i db
```

CSV: https://gist.github.com/rfdavid/ba4d28cbcd2c727794fe59af4adc74ef

**Comment by acquamarin** [comment]
Hi Rui,

I tried to load the csv you provided and query the `Preference` table. However, i can't reproduce the error you reported. Could you please provide me your building environment so we can better dig into this issue?

Thanks,

Ziyi

**Comment by rfdavid** [comment]
Hi Ziyi,

I built both on debug and release, and basically ran the following
```
DROP table Preference; 
CREATE NODE TABLE Preference(name STRING, pref STRING[],  PRIMARY KEY (name)) ; 
COPY Preference FROM "pref.csv"; 
match (p:Preference) return p.pref;
```

In release (both compiling and downloading from the website), the query runs normally and doesn't produce any seg fault. I'm running on Apple Silicon. Is there anything more specific that I can provide?

Rui

**Comment by acquamarin** [comment]
Hi Rui,

Thanks for you feedback, I could reproduce the bug under Apple Silicon and DEBUG mode. We will fix this bug as soon as possible.

Ziyi

**Comment by acquamarin** [comment]
Hi Rui,

This bug is caused by copying an empty list and i have fixed it in PR #1358. Can you try to load your csv again and confirm whether the bug has been solved.

**Comment by rfdavid** [comment]
I compiled it again and everything is working now 👍
